x-pack/plugin/enrich/src/main/java/org/elasticsearch/xpack/enrich/EnrichPolicyExecutor.java

```diff
@@ -115,26 +129,28 @@ private void tryLockingPolicy(String policyName) {
                     + "]"
             );
         }
+        return policyLock;
     }
 
-    private void releasePolicy(String policyName) {
+    private void releasePolicy(EnrichPolicyLock policyLock) {
```

Comment:
What's the difference between `releasePolicy(policyLock)` and `policyLock.release()`? Is there a way we can differentiate the two a bit better? It looks like the main difference is the just the execution permits?

Reply:
I could update the name to make it a bit more clear - but yes, the cleanup logic here needs us to release a permit as well as unlock the policy.

Reply:
I changed the policy lock object to implement Releasable, and now I simply have the method wrap the Releasable so that it also releases the execution permits. Everything just calls `close()` now instead of needing to remember to run `releasePolicy(policyLock)`